### PR TITLE
Remove [Conformance] from tests in test/e2e_node

### PR DESCRIPTION
**Pull request by spiffxp**
Conformance tests live inside of test/e2e, none of the tests currently
tagged as `[Conformance]` in test/e2e_node actually get run when you run
the conformance tests with e2e.test (either directly or indirectly with
sonobuoy)

If these tests make sense as both `[NodeConformance]` and `[Conformance]`
tests, they should be ported to test/e2e/common

/kind cleanup
/area conformance
/sig architecture
/cc @bgrant0607 @smarterclayton @timothysc
/sig node
/cc @yujuhong

ref: https://github.com/kubernetes/kubernetes/issues/66875

```release-note
NONE
```

### CODE REVIEW

File: test/conformance/testdata/conformance.txt [hunk 1]
```diff
@@ -176,13 +176,3 @@ test/e2e/storage/subpath.go: "should support subpaths with configmap pod"
 test/e2e/storage/subpath.go: "should support subpaths with configmap pod with mountPath of existing file"
 test/e2e/storage/subpath.go: "should support subpaths with downward pod"
 test/e2e/storage/subpath.go: "should support subpaths with projected pod"
-test/e2e_node/kubelet_test.go: "it should print the output to logs"
-test/e2e_node/kubelet_test.go: "it should not write to root filesystem"
-test/e2e_node/lifecycle_hook_test.go: "should execute poststart exec hook properly"
```
Comment (spiffxp) [inline test/conformance/testdata/conformance.txt]: I am pretty sure these lifecycle tests _should_ be conformance

File: test/e2e_node/kubelet_test.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ var _ = framework.KubeDescribe("Kubelet", func() {
 			Testname: Kubelet, log output, default
 			Description: By default the stdout and stderr from the process being executed in a pod MUST be sent to the pod's logs.
 		*/
-		framework.ConformanceIt("it should print the output to logs [NodeConformance]", func() {
+		It("it should print the output to logs [NodeConformance]", func() {
```
Comment (timothysc) [inline test/e2e_node/kubelet_test.go]: this file should be ported.

File: test/e2e_node/lifecycle_hook_test.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
 			Testname: Pod Lifecycle, post start exec hook
 			Description: When a post start handler is specified in the container lifecycle using a ‘Exec’ action, then the handler MUST be invoked after the start of the container. A server pod is created that will serve http requests, create a second pod with a container lifecycle specifying a post start that invokes the server pod using ExecAction to validate that the post start is executed.
 		*/
-		framework.ConformanceIt("should execute poststart exec hook properly [NodeConformance]", func() {
+		It("should execute poststart exec hook properly [NodeConformance]", func() {
```
Comment (timothysc) [inline test/e2e_node/lifecycle_hook_test.go]: this file should be ported.

File: test/e2e_node/runtime_conformance_test.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ var _ = framework.KubeDescribe("Container Runtime Conformance Test", func() {
 				Testname: Container Runtime, restart policy
 				Description:  If the restart policy is set to ‘Always’, Pod MUST be started when terminated, If restart policy is ‘OnFailure’, Pod MUST be started only if it is terminated with non-zero exit code. If the restart policy is ‘Never’, Pod MUST never be restarted. All these three test cases MUST verify the restart counts accordingly.
 			*/
-			framework.ConformanceIt("it should run with the expected status [NodeConformance]", func() {
+			It("it should run with the expected status [NodeConformance]", func() {
```
Comment (timothysc) [inline test/e2e_node/runtime_conformance_test.go]: this file should be ported.

**Review by timothysc — APPROVED** [review]
I'm ok with the change if there is an issue opened to port the marked ones to the conformance suite. 

/approve

File: test/e2e_node/runtime_conformance_test.go (hunk 4 above)
Comment (spiffxp) [reply test/e2e_node/runtime_conformance_test.go]: #67103 (this seems less obvious to me, but let's see how far we can gert)

File: test/e2e_node/kubelet_test.go (hunk 2 above)
Comment (spiffxp) [reply test/e2e_node/kubelet_test.go]: #67104

**Comment by bgrant0607** [comment]
Oh, these were added by #54837, probably before we had the OWNERS file put in place to prevent additions without review

File: test/e2e_node/kubelet_test.go (hunk 2 above)
Comment (bgrant0607) [reply test/e2e_node/kubelet_test.go]: We would need to figure out how to get the logs in a portable, reliable fashion.

File: test/e2e_node/kubelet_test.go [hunk 5]
```diff
@@ -174,7 +174,7 @@ var _ = framework.KubeDescribe("Kubelet", func() {
 			Testname: Kubelet, Pod with read only root file system
 			Description: Create a Pod with security context set with ReadOnlyRootFileSystem set to true. The Pod then tries to write to the /file on the root, write operation to the root filesystem MUST fail as expected.
 		*/
-		framework.ConformanceIt("it should not write to root filesystem [NodeConformance]", func() {
+		It("it should not write to root filesystem [NodeConformance]", func() {
```
Comment (bgrant0607) [inline test/e2e_node/kubelet_test.go]: Pod feature. Should be covered by conformance

File: test/e2e_node/gke_environment_test.go [hunk 6]
```diff
@@ -310,7 +310,7 @@ func checkDockerStorageDriver() error {
 	return fmt.Errorf("failed to find storage driver")
 }
 
-var _ = framework.KubeDescribe("GKE system requirements [Conformance][NodeConformance][Feature:GKEEnv][NodeFeature:GKEEnv]", func() {
+var _ = framework.KubeDescribe("GKE system requirements [NodeConformance][Feature:GKEEnv][NodeFeature:GKEEnv]", func() {
```
Comment (bgrant0607) [inline test/e2e_node/gke_environment_test.go]: The purpose of this was to just dump info to the test log?

File: test/e2e_node/mirror_pod_test.go [hunk 7]
```diff
@@ -62,7 +62,7 @@ var _ = framework.KubeDescribe("MirrorPod", func() {
 			Testname: Mirror Pod, update
 			Description: Updating a static Pod MUST recreate an updated mirror Pod. Create a static pod, verify that a mirror pod is created. Update the static pod by changing the container image, the mirror pod MUST be re-created and updated with the new image.
 		*/
-		framework.ConformanceIt("should be updated when static pod updated [NodeConformance]", func() {
+		It("should be updated when static pod updated [NodeConformance]", func() {
```
Comment (bgrant0607) [inline test/e2e_node/mirror_pod_test.go]: Mirror pod tests do not belong in conformance. Static pods require reconfiguring kubelets, which make it not an API feature.

File: test/e2e_node/runtime_conformance_test.go (hunk 4 above)
Comment (bgrant0607) [reply test/e2e_node/runtime_conformance_test.go]: Container restart behavior is a portable pod feature, so it should be covered by conformance.

File: test/e2e_node/runtime_conformance_test.go [hunk 8]
```diff
@@ -132,7 +132,7 @@ while true; do sleep 1; done
 					By("it should get the expected 'State'")
 					Expect(GetContainerState(status.State)).To(Equal(testCase.State))
 
-					By("it should be possible to delete [Conformance][NodeConformance]")
+					By("it should be possible to delete [NodeConformance]")
```
Comment (bgrant0607) [inline test/e2e_node/runtime_conformance_test.go]: It's weird that this is a subpart of this test case, or sequence of test cases. I'm not sure it's right even for NodeConformance.

File: test/e2e_node/runtime_conformance_test.go [hunk 9]
```diff
@@ -147,7 +147,7 @@ while true; do sleep 1; done
 				message   gomegatypes.GomegaMatcher
 			}{
 				{
-					name: "if TerminationMessagePath is set [Conformance][NodeConformance]",
+					name: "if TerminationMessagePath is set [NodeConformance]",
```
Comment (bgrant0607) [inline test/e2e_node/runtime_conformance_test.go]: TerminationMessagePath is a pod feature that should be covered by conformance.

File: test/e2e_node/runtime_conformance_test.go [hunk 10]
```diff
@@ -318,7 +318,7 @@ while true; do sleep 1; done
 				},
 			} {
 				testCase := testCase
-				It(testCase.description+" [Conformance][NodeConformance]", func() {
+				It(testCase.description+" [NodeConformance]", func() {
```
Comment (bgrant0607) [inline test/e2e_node/runtime_conformance_test.go]: We'll want to test image pulling and image pull secrets (but not host-level secret configuration), but will need to make the registry pulled from configurable.

**Review by bgrant0607 — COMMENTED** [review]
Thanks!
I left comments regarding whether specific tests would make sense as part of conformance or not.
/lgtm
/approve

**Pull request merged**